src/main/kotlin/snyk/common/lsp/SnykLanguageClient.kt

```diff
@@ -120,33 +120,37 @@ class SnykLanguageClient :
 
         //If the diagnostics for the file is empty, clear the cache.
         if (firstDiagnostic == null) {
-            scanPublisher.onPublishDiagnostics("code", snykFile, emptyList())
-            scanPublisher.onPublishDiagnostics("oss", snykFile, emptyList())
+            clearCache(scanPublisher, snykFile)
             return
         }
 
+        val issueList = getScanIssues(diagnosticsParams)
+        updateCacheForProduct(scanPublisher, product, snykFile, issueList)
+
+        return
+    }
+
+    fun getScanIssues(diagnosticsParams: PublishDiagnosticsParams): List<ScanIssue> {
         val issueList = diagnosticsParams.diagnostics.stream().map {
             Gson().fromJson(it.data.toString(), ScanIssue::class.java)
         }.toList()
+        return issueList
+    }
 
-        when (product) {
-            LsProductConstants.OpenSource.value -> {
-                scanPublisher.onPublishDiagnostics(product, snykFile, issueList)
-            }
-
-            LsProductConstants.Code.value -> {
-                scanPublisher.onPublishDiagnostics(product, snykFile, issueList)
-            }
-
-            LsProductConstants.InfrastructureAsCode.value -> {
-                // TODO implement
-            }
+    fun clearCache(scanPublisher: SnykScanListenerLS, snykFile: SnykFile) {
+        updateCacheForProduct(scanPublisher, "code", snykFile, emptyList())
+        updateCacheForProduct(scanPublisher, "oss", snykFile, emptyList())
+    }
```

Comment:
I would not call this clearCache, it should be sth like `publishEmptyDiagnostics`. Using the event bus is meant to decouple event producers from event consumers. We should not assume that a cache is consuming the event.